mdm/server.go

```diff
@@ -48,29 +49,36 @@ func RegisterHTTPHandlers(r *mux.Router, e Endpoints, verifier SignatureVerifier
 }
 
 type SignatureVerifier interface {
-	VerifySignature(sig string, message []byte) error
+	VerifySignature(sig string, message []byte) (*x509.Certificate, error)
 }
 
 type requestDecoder struct {
 	verifier SignatureVerifier
 }
 
-func (d *requestDecoder) readBody(r *http.Request) ([]byte, error) {
+func (d *requestDecoder) readBody(r *http.Request) ([]byte, *x509.Certificate, error) {
 	defer r.Body.Close()
 
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		return nil, errors.Wrap(err, "reading MDM Response HTTP Body")
+		return nil, nil, errors.Wrap(err, "reading MDM Response HTTP Body")
 	}
 
+	// TODO: If we ever use Go client cert auth we can use
```

Comment:
One thing to be cautious of here is that if we use TLS auth, we cannot reject via the handler. It has to be done at the TLS layer, before the handler has a chance to work with the request.

Reply:
That's fine. As long as we shut down the rest of processing so a client can't access the command queue (through an HTTP error) that should do it.